src/status_im/chat/models/images.cljs

```diff
@@ -23,6 +24,14 @@
       (.then #(on-success (.path %)))
       (.catch #(log/error "could not save image"))))
 
+(defn share-image
+  [uri]
+  (download-image-http uri
```

Comment:
@OmarBasem, I know this is not your PR's responsibility, but I need to say the usage of promises in `share-image`, `download-image-http` and `share/open` are basically incorrect by mixing callbacks with promises. There's a serious misunderstanding of promises in status-mobile.

Async functions don't need to receive callbacks and they can be linearized instead of nested. I hope to create a PR soon refactoring and then I'll mark some devs as reviewers to share knowledge.

Reply:
Thanks for the feedback @ilmotta! Can you please give an example of how to write it linearized?

Reply:
Glad to share @OmarBasem :)

Coming up with examples would take some time now, but this article is a good primer https://plainenglish.io/blog/nested-promise-an-anti-pattern-alternative-to-it. We basically do promise nesting and use callbacks where we don't need any. In this PR `share-image` does exactly this.

I'm going to start a branch on using Clojure's promesa library, and see what it gives, let's see :shrug: I used it with another team in ClojureScript and it was quite easy for most people to learn and it gives us some syntactic sugar. It's also a solid, stable implementation for a long time already.

Reply:
Thanks for the explanation Icaro!